### PR TITLE
feat(tasks,simulators): feature branch

### PR DESCRIPTION
**Pull request by amitlissack**
This branch combines the following PRs:
#187 
#191 
#196
#198 

Big Features:
- CAN messages are written to FIFO from a dedicated FreeRTOS task.
- New FreeRTOS tasks:
      - Motor driver task for communicating with MotorDriver class.
      - Motion controller task for MotionController class.
      - Move group task for accessing and executing a move group.
      - EEProm task (pipettes only) for access to eeprom
- Head and Pipettes simulators build out.

### CODE REVIEW

File: head/firmware/main.cpp [hunk 1]
```diff
@@ -6,19 +6,171 @@
...
+    .enable = {
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+        .port = GPIOB,
+        .pin = GPIO_PIN_11,
+        .active_setting = GPIO_PIN_SET},
+};
+
+motor_driver_config::RegisterConfig MotorDriverConfigurations{
+    .gconf = 0x04,
+    .ihold_irun = 0x70202,
```
Comment (sfoster1) [inline head/firmware/main.cpp:102]:
```suggestion
    .ihold_irun = 0x71002,
```

Comment (sfoster1) [reply head/firmware/main.cpp:102]: (bumps up the current when active)

File: head/core/can_task.cpp [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    // Accept head left
+    filter.node_id(can_ids::NodeId::head_l);
+    can_bus->add_filter(CanFilterType::mask, CanFilterConfig::to_fifo1, filter,
+                        can_arbitration_id::ArbitrationId::node_id_bit_mask);
+
+    // Reject everything else.
+    can_bus->add_filter(CanFilterType::mask, CanFilterConfig::reject, 0, 0);
+
+    // TODO (al, 2021-12-21): Must account for this!!
+    //    motor_left.driver.setup();
```
Comment (sfoster1) [inline head/core/can_task.cpp:171]: noting this quick - is this a leftover comment or are these in fact never called?

**Review by sfoster1 — COMMENTED** [review]
Some quick details

File: head/firmware/main.cpp (hunk 1 above)
Comment (amitlissack) [reply head/firmware/main.cpp:102]: Is this a merge error?

File: head/core/can_task.cpp (hunk 2 above)
Comment (amitlissack) [reply head/core/can_task.cpp:171]: It should be deleted.

File: head/firmware/main.cpp (hunk 1 above)
Comment (amitlissack) [reply head/firmware/main.cpp:102]: Sure looks like it.

**Review by Laura-Danielle — APPROVED** [review]

**Review by sfoster1 — APPROVED** [review]
LGTM modulo the above small changes

**Pull request merged**